lms/djangoapps/grades/course_grade_factory.py

```diff
@@ -193,26 +197,27 @@ def _update(user, course_data, force_update_subsections=False):
                 passed=course_grade.passed,
             )
 
-        COURSE_GRADE_CHANGED.send_robust(
-            sender=None,
-            user=user,
-            course_grade=course_grade,
-            course_key=course_data.course_key,
-            deadline=course_data.course.end,
-        )
-        if course_grade.passed:
-            COURSE_GRADE_NOW_PASSED.send(
-                sender=CourseGradeFactory,
-                user=user,
-                course_id=course_data.course_key,
-            )
-        else:
-            COURSE_GRADE_NOW_FAILED.send(
-                sender=CourseGradeFactory,
+        if send_course_grade_signals:
```

Comment:
It is the case, that every time that `send_course_grade_signals` is false, `should_persist_grades` is also false?

I'm thinking that if signals should not be sent when persistent grades is disabled, then we could make the if statement dependant on that variable and avoid creating a new control variable that needs to be accounted for and passed by the generation_handler.

Reply:
We should send signals 1 time regardless of the persistent grade status.

The flow is:
- user answers the question in the course;
- recalculation of course grade is starting;
- course grade now passing -> send the `COURSE_GRADE_NOW_PASSED` signal to start certificate generation;

*here we should not break the signal sending*

- certificate generation process would like to receive the course grades so it calls the `course_grade_factory` again

*here we should avoid sending the signals*

Reply:
This makes sense. we should not send the signal within the processing of the same signal.

Since this was the last pending thing, I'll leave my approve